src/test/HostActivationTests/FrameworkResolution/RollForwardMultipleFrameworks.cs

```diff
@@ -0,0 +1,576 @@
+            string versionReference,
+            string rollForward,
+            string resolvedFramework)
+        {
+            RunTest(
+                runtimeConfig => runtimeConfig
+                    .WithFramework(MicrosoftNETCoreApp, "5.1.1")
+                    .WithFramework(MiddleWare, "2.1.0"),
+                dotnetCustomizer => dotnetCustomizer.Framework(MiddleWare).RuntimeConfig(runtimeConfig =>
+                    runtimeConfig.GetFramework(MicrosoftNETCoreApp)
+                        .WithRollForward(rollForward)
+                        .Version = versionReference))
+                .ShouldHaveResolvedFrameworkOrFailedToReconcileFrameworkReference(
+                    MicrosoftNETCoreApp, resolvedFramework, "5.1.1", versionReference);
+        }
+
+        // Verify that inner framework reference (<fxRefVersion>, <rollForward>)
+        // is correctly reconciled with app's framework reference 6.1.1-preview.0 <defaults>.
+        // Also validates the effect of DOTNET_ROLL_FORWARD_TO_PRERELEASE on the result.
+        [Theory] // fxRefVersion       rollForward                               rollForwadToPreRelease resolvedFramework
+        [InlineData("6.0.0-preview.1", null,                                     false,                 "6.1.1-preview.2")]
+        [InlineData("6.0.0",           null,                                     false,                 "6.2.1")]
+        [InlineData("6.0.0",           Constants.RollForwardSetting.LatestPatch, false,                 null)]
+        [InlineData("6.0.0-preview.1", Constants.RollForwardSetting.LatestPatch, false,                 null)]
+        [InlineData("6.0.0-preview.1", Constants.RollForwardSetting.Minor,       false,                 "6.1.1-preview.2")]
+        [InlineData("6.0.0",           Constants.RollForwardSetting.Minor,       false,                 "6.2.1")]
+        [InlineData("6.0.1-preview.0", Constants.RollForwardSetting.LatestPatch, false,                 null)]
+        [InlineData("6.1.0-preview.0", null,                                     false,                 "6.1.1-preview.2")]
+        [InlineData("6.1.0-preview.0", null,                                     true,                  "6.1.1-preview.2")]
+        [InlineData("6.1.0",           null,                                     false,                 "6.2.1")]
+        [InlineData("6.1.0",           null,                                     true,                  "6.1.1-preview.2")]
+        [InlineData("6.1.1-preview.0", null,                                     false,                 "6.1.1-preview.2")]
+        [InlineData("6.1.1-preview.0", null,                                     true,                  "6.1.1-preview.2")]
+        [InlineData("6.1.1-preview.0", Constants.RollForwardSetting.LatestPatch, false,                 "6.1.1-preview.2")]
+        [InlineData("6.1.1-preview.0", Constants.RollForwardSetting.Disable,     false,                 "[not found]")]
```

Comment:
`[not found]` ? Seems like this should be `null` to be consistent.

Reply:
The test can have basically 3 outcomes:
* It did resolve some framework   - "a.b.c"
* It failed to reconcile the two framework references  - null
* It did reconcile the framework references, but the resulting framework reference could not be resolved against the available frameworks.  - "[not found]"

It is a bit confusing, but I can't think of a better way to do this.

Reply:
I found this confusing as well (figuring out the distinction required looking at the assert / reasoning backwards for me). Maybe comment of what you said above and/or define constants with names more indicative of 'failed to reconcile' vs 'reconciled, not available'

Reply:
I introduced named constants instead of the `null` which should make this much more easy to read.